snapcraft/internal/elf.py

```diff
@@ -48,18 +48,61 @@ def __init__(self, *, path: str, is_executable: bool) -> None:
         """
         self.path = path
         self.is_executable = is_executable
+        self.dependencies = set()  # type: Set[str]
+
+    def load_dependencies(self) -> Set[str]:
+        """Load the set of libraries that are needed to satisfy elf's runtime.
+
+        This may include libraries contained within the project.
+        The object's .dependencies attribute is set after loading.
```

Comment:
What about making this a lazy property instead of a side-effect?
Make _load_dependencies internal, and when ElfFile.dependencies is called the first time, it will load the dependencies. The following times, it will just return the loaded value.

Reply:
this func/method will die in the following PRs, I'd rather not do it.

Reply:
We also, in its current form would want to reload (that can be solved with a new method)